webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactRepository.scala

```diff
@@ -190,12 +190,25 @@ trait NodeFactRepository {
    * Save (create or override) a core node fact
    * Use "updateInventory` if you want to save in pending, it's likely what you want.
    *
-   * Not that the diff is only done on the core properties
+   * Not that the diff is only done on the core properties.
+   * SecurityTag must not be updated with that method, but it can be initially set.
    */
   def save(
       nodeFact:  NodeFact
   )(implicit cc: ChangeContext, attrs: SelectFacts = SelectFacts.all): IOResult[NodeFactChangeEventCC]
 
+  /*
+   * A simpler version of save for CoreNodeFacts
+   */
+  def save(nodeFact: CoreNodeFact)(implicit cc: ChangeContext): IOResult[NodeFactChangeEventCC] = {
+    save(NodeFact.fromMinimal(nodeFact))(cc, SelectFacts.none)
+  }
+
+  /*
+   * Define security tag for the node if change context allows it.
+   */
+  def setSecurityTag(nodeId: NodeId, tag: Option[SecurityTag])(implicit cc: ChangeContext): IOResult[NodeFactChangeEventCC]
```

Comment:
:star: this is the important change for updating node security context (ie in what tenant it belongs).  
Since the security context is part of the `NodeFact`, we must tightly control how it can be changed. 
For that, we chose to have: 
- the `save` method isn't allowed to change security context, 
- there is a dedicated method for that.